src/Microsoft.Health.Fhir.Core/Features/Operations/Export/CreateExportRequestHandler.cs

```diff
@@ -179,5 +205,96 @@ private ExportJobFormatConfiguration ParseFormat(string formatName, bool useCont
+                return;
+            }
+
+            var errors = new List<string[]>();
+            foreach (var filter in filters)
+            {
+                if (filter.Parameters == null || filter.Parameters.Count == 0)
+                {
+                    continue;
+                }
+
+                foreach (var parameter in filter.Parameters)
+                {
+                    string name = null;
+                    string modifier = null;
+                    if (!TrySplitParameterName(parameter.Item1, out name, out modifier)
+                        || string.IsNullOrWhiteSpace(name)
+                        || (!string.IsNullOrWhiteSpace(modifier) ? !KnownSearchParameterModifiers.Contains(modifier) : false))
+                    {
+                        errors.Add(new string[] { filter.ResourceType, parameter.Item1, "Unknown" });
+                        continue;
+                    }
+
+                    // NOTE: some search parameters loaded into SearchParameterDefinitionManager's cache from the embedded resource (search-parameters.json)
+                    //       have the status with "0", inconsistent with the status in the data store. Allow the "0" status as Enabled for now.
+                    SearchParameterInfo searchParameter = null;
+                    if (_searchParameterDefinitionManager.TryGetSearchParameter(filter.ResourceType, name, out searchParameter)
+                        && (searchParameter.SearchParameterStatus == SearchParameterStatus.Enabled
+                        || searchParameter.SearchParameterStatus == 0))
+                    {
+                        continue;
+                    }
+
+                    errors.Add(new string[]
+                        {
+                            filter.ResourceType,
+                            parameter.Item1,
+                            searchParameter?.SearchParameterStatus.ToString() ?? "Unknown",
+                        });
+                }
+            }
+
+            if (errors.Count > 0)
+            {
+                var errorMessage = new StringBuilder($"{errors.Count} invalid search parameter(s) found:{Environment.NewLine}");
+                errors.ForEach(e => errorMessage.AppendLine(
+                    string.Format(CultureInfo.InvariantCulture, "[type: {0}, parameter: {1}, status: {2}]", e[0], e[1], e[2])));
+
+                var message = errorMessage.ToString();
+                _logger.LogError(message);
+
+                var ex = new BadRequestException(message);
+                if (_includeValidateTypeFiltersValidationDetails)
+                {
+                    // Note: Test purpose only
+                    ex.Data.Add(nameof(ValidateTypeFilters), errors);
+                }
+
+                throw ex;
+            }
+        }
+
+        private static bool TrySplitParameterName(string parameterName, out string name, out string modifier)
```

Comment:
This won't work with reverse chained search parameters and I don't think it will work as intended with chained search parameters. https://www.hl7.org/fhir/R4/search.html#chaining

Reply:
Fixed by switching to using ISearchOptionFactory.Create that handles chained search parameters.